Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM nginx:alpine
+
+RUN apk --no-cache add nodejs-lts git
+
+WORKDIR /tmp/app
+ADD . ./
+
+RUN npm install -g yarn bower
+RUN yarn
+RUN bower --allow-root install
+RUN yarn run build optimize-images
+RUN yarn run build
+
+COPY ./build/bundled/. /usr/share/nginx/html
+
+RUN rm -r /tmp/app
+RUN apk del nodejs-lts git
```

Comment:
Een Docker image bestaat uit verschillende layers die over elkaar worden gelegd. `docker build` maakt voor elk commando (dus ook elke `RUN`) een layer aan. Die layers worden niet samengevoegd (dat is nog een experimentele feature) wat betekent dat als je in layer `n` iets toevoegt en in layer `n+1` weer verwijdert de bestanden nog steeds deel uit maken van je image (en dat is dus ruimteverspilling). 

Om dit op te lossen kun je alle commando's in één `RUN` doen door ze met `&&` aan elkaar te plakken. Zie bijvoorbeeld https://github.com/praseodym/docker-znc-armhf/blob/master/Dockerfile waarbij de build dependencies worden geïnstalleerd en in dezelfde layer nog worden verwijderd, zodat ze geen deel uitmaken van het image.